### PR TITLE
Log bucket policy edits

### PR DESCRIPTION
**Pull request by matt-heery**

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -255,3 +255,64 @@ data "aws_iam_policy_document" "default" {
...
+locals {
+  new_policy_statements = {
+    for bucket_key, bucket in var.log_buckets : bucket_key => {
+      Sid    = "AllowS3Logging"
+      Effect = "Allow"
+      Principal = {
+        Service = "logging.s3.amazonaws.com"
+      }
+      Action   = "s3:PutObject"
+      Resource = "arn:aws:s3:::${bucket.id}/*"
```
Comment (pricemg) [inline main.tf:274]: could you just do bucket.arn here?

**Review by pricemg — APPROVED** [review]

File: main.tf (hunk 1 above)
Comment (matt-heery) [reply main.tf:274]: good point

File: variables.tf [hunk 2]
```diff
@@ -116,7 +116,8 @@ variable "lifecycle_rule" {
 
 variable "log_buckets" {
   type = map(object({
-    id = string
+    id  = string
+    arn = string
```
Comment (pricemg) [inline variables.tf:120]: (Y)

**Review by pricemg — APPROVED** [review]

**Review by dms1981 — APPROVED** [review]

**Comment by dms1981** [comment]
LGTM - really appreciate the work you put in on the tests!

**Pull request merged**